tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala

```diff
@@ -1928,6 +1928,28 @@ class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
     }
   }
 
+  test("#2774 1") {
+    val tree =
+      Term.Param(Mod.Erased() :: Mod.Using() :: Nil, Term.Name("foo"), Some(Type.Name("Bar")), None)
+    assertSyntax(tree, "erased foo: Bar")(tree)
+  }
+
+  test("#2774 2") {
+    val tree = Type.Function(
+      List(Type.TypedParam(Type.Name("e"), Type.Name("Entry"), List(Mod.Erased()))),
+      Type.Select(Term.Name("e"), Type.Name("Key"))
+    )
+    assertSyntax(tree, "(erased e: Entry) => e.Key")(tree)
+  }
+
+  test("#2774 3") {
+    val tree = Type.Function(
+      List(Type.FunctionArg(List(Mod.Erased()), Type.Name("Ev")), Type.Name("Int")),
```

Comment:
Actually do we even need `FunctionArg` Shouldn't it be List(Type.TypedParam(Type.Name.Anonymous, Type.Name("Entry"), List(Mod.Erased()))), maybe?

Feels like `Type.FunctionArg` is just for mods, which is a bit of a inconsistency :thinking:

Reply:
Ach, that's fine then. Thanks for explaning!